src/Controller/UserController.php

```diff
@@ -145,4 +144,53 @@ public function deleteUser($id, UserRepository $userRepository): JsonResponse
 
         return new JsonResponse(['message' => 'The user was successfully deleted!'], Response::HTTP_OK);
     }
+
+    /**
+     * Change password
+     * @Rest\Post("/{id}/change_password")
+     * @param User $user
+     * @param UserRepository $userRepository
+     * @param Request $request
+     * @return JsonResponse
+     * @throws ORMException
+     * @throws OptimisticLockException
+     */
+    public function userChangePassword(
+        User $user,
+        UserRepository $userRepository,
+        Request $request
+    ): JsonResponse {
+        $authenticatedUser = $this->getUser();
+
+        if ($authenticatedUser->getId() !== $user->getId()) {
+            return new JsonResponse(['message' => 'Access denied!'], Response::HTTP_BAD_REQUEST);
```

Comment:
Actualy it should be Response::HTTP_FORBIDDEN as the current user cannot edit this user. You can fix it after the merge.